lib/crawler/dsl/configuration.rb

```diff
@@ -0,0 +1,28 @@
+module Crawler
+  module Dsl
+    class Configuration
+      include DslOrganizer::ExportCommand[:configuration]
+
+      OPTIONS = [:screenshots_path, :max_pages, :window_width, :window_height]
+
+      DEFAULT_OPTIONS = {
+          window_width: 1280,
+          window_height: 1600
+      }
+
+      def call
+        yield(self)
+      end
+
+      def options
+        @options ||= DEFAULT_OPTIONS
+      end
+
+      OPTIONS.each do |option|
```

Comment:
1. Configuration is probably should not go under DSL, as it configures Crawler itself. So just `Crawler::Configuration`

2. Do we really need using metaprogramming to define methods? 
Metaprog is ok, just statically defined methods are easier to find, read and maintain.

Similar Configuration approach is implemented in Capybara gem:

- all configuration methods are defined in a separate class Capybara::Config (we will have Crawler::Config)
- Crawler contains private(or public if you wish) method `config` that instantiates the configuration
```ruby
  def config
     config = Crawler::Config.new
 end
```
- configuration is made with 
```ruby
    def configure
      yield config
    end
```
- It is possible to expose configuration methods on the Crawler class itself with the method delegation:

Reply:
Configuration example:
1. Existing, pass config as constructor parameters :
```
crawler = Crawler::Engine.new(max_pages: 1)
```

2. Pass config as a hash
```
crawler = Crawler::Engine.new(config)
```

3. Configure an instance
```
crawler = Crawler::Engine.new
# ... if we need to create crawler separately from configuration
crawler.configure do |config|
  config.max_width = 155;
end
```

4. Global config
```
Crawler::Engine.configure do | config |
  config.screenshot_path = './tmp/screenshots'
end
```